### DISCUSSION TITLE
How to disable the $DISPLAY on the X11?

### DISCUSSION THREAD

**Discussion by kimcoldwater** [FreeRDP/FreeRDP#8089] [Q&A]
Hello.

I'm sorry to post a series of questions.

I am creating RDP functions on the web using dynamic libraries and JNI that are created after FreeRDP builds. ( like Android version)

How can I use the X11 without the $DISPLAY environment variable?

I don't need a display, I only need the screen data that is delivered to EndPaint.

We've looked at `xfreerdp_client_new` and it looks like `*Display` structure needs to be implemented.

Could you advise me which direction I should change the source to run the app normally without the DISPLAY environment variable on the X11?

Do I just need to remove the $DISPALY environmental variable exception and put any value (corresponding to the display) in the structure?

Thank you for reading the question.

**Comment by mfleisz** [comment]
You cannot use the X11 client without a running a X-Server - after all it is a X11 client.

If you need a headless client you'll need to create your own client. There is a sample client (which can be found in client/sample) that you can use as a good starting point.

**Reply by kimcoldwater** [reply]
Ok.

I know how to use samples while customizing FreeRDP using existing clients for several months.

But I don't understand Cmake well because I mainly used java and js.

If I look at the CMakeLists.txt of the sample from the existing client, there are many differences.

As I said, I just need EndPaint's screen data.

Can I create a headless client without any problems by creating a client according to the sample and using 'CMakeLists.txt' in the sample?

I know how to customize existing CMakeLists, but I don't know which one to add.

I'm going to use this in mac and linux.
